### ISSUE TITLE
Keep permissions

### ISSUE THREAD

**Issue by reznikmm** [alire-project/alire#1763]
**Describe the bug**
alr 2.1.x resets permissions on making a shared copy of a release.

As result `post-fetch` action can't execute bash scripts.

**To Reproduce**
1. alr settings --global --set dependencies.shared true
2. alr get gtkada
3. alr get --build spawn_glib=23
4. Error is:

> /usr/bin/bash: рядок 1: ./configure: Permission denied
> error: Command ["bash", "-c", "PATH=${DISTRIB_ROOT}/mingw64/bin:${PATH} ./configure"] exited with code 126

**Expected Behavior**
It looks 
Describe what happened and how it deviates from what should have happened, if unclear from the previous steps.

**`alr` version**
```
APPLICATION
alr version:               2.1.0-dev+62b73c9
libalire version:          2.1.0-dev+62b73c9
compilation date:          2024-09-24 00:08:06
compiled with version:     14.2.0

CONFIGURATION
settings folder:           /home/max/.config/alire
cache folder:              /home/max/.local/share/alire
vault folder:              /home/max/.local/share/alire/releases
build folder:              /home/max/.local/share/alire/builds
temp folder:               /run/user/1000
force flag:                FALSE
non-interactive flag:      FALSE
community index branch:    stable-1.3.0
compatible index versions: ^1.1 & <=1.3.0
indexes folder:            /home/max/.config/alire/indexes
indexes metadata:          OK
index #5:                  (community) git+https://github.com/alire-project/alire-index#stable-1.3.0
toolchain folder:          /home/max/.local/share/alire/toolchains
toolchain assistant:       disabled
tool #1 gnat:              gnat_native=14.2.1
tool #2 gprbuild:          gprbuild=22.0.1
system package manager:    /usr/bin/apt
distro detection disabled: FALSE

WORKSPACE
root status:               BROKEN
root release:              N/A
root load error:           Uninitialized Outcome
root folder:               N/A
current folder:            /home/max/.local/share/alire/builds/gtkada_24.0.0_80c56171/05561da954afaee17128f59bba699db2f2e4a4215317b2b818a8016192c82334

SYSTEM
distribution:              UBUNTU
host-arch:                 X86_64
os:                        LINUX
target:                    NATIVE
toolchain:                 SYSTEM
word-size:                 BITS_64
```

**`alr` logs**
```
debug: Entering folder: /home/max/.local/share/alire/builds/gtkada_24.0.0_80c56171/05561da954afaee17128f59bba699db2f2e4a4215317b2b818a8016192c82334
info: ⓘ Running post-fetch actions for gtkada=24.0.0...
detail: Running action: Post_Fetch run: bash -c PATH=${DISTRIB_ROOT}/mingw64/bin:${PATH} ./configure (from ${CRATE_ROOT}/.) (cwd:/home/max/.local/share/alire/builds/gtkada_24.0.0_80c56171/05561da954afaee17128f59bba699db2f2e4a4215317b2b818a8016192c82334)
debug: Entering folder: .
detail: Spawning: ["bash", "-c", "PATH=${DISTRIB_ROOT}/mingw64/bin:${PATH} ./configure"]
```

If I look at that directory then there is no `x` permission:
```shell
cd /home/max/.local/share/alire/builds/gtkada_24.0.0_80c56171/05561da954afaee17128f59bba699db2f2e4a4215317b2b818a8016192c82334
ls -l configure
-rw-rw-r-- 1 max max 162964 Sep 24 15:58 configure
```
If pertinent, paste the output of `alr -vv -d <your command>` here.

**Comment by mosteo** [comment]
Thanks for the report.

**Comment by mosteo** [comment]
Fixed in #1768
